DateModule.js

```diff
@@ -0,0 +1,190 @@
+//coder: Katja Borchert
+
+//A Date Utility Module in the "Revealing Pattern Module" format
+/* eslint-disable no-unused-vars */
+var myObj = (function iife() {
+/* eslint-enable no-unused-vars */
+
+  //private variables:
+  var date = null;
+  
+  //private function: date instantiation
+  function _initDate(){
+    date = new Date();
+  }
+    
+  //public API:
+  
+  //setDate(): sets the internal date based on user input
+  //takes in: empty argument list, date object, or number)
+  function setDate(enteredDate) {
+  
+    //check that date is set: if not, initialize date to now
+    if (date === null) {
+      _initDate();
+    }
+	
+	/*checks the type of the parameter enteredDate:
+	check if the enteredDate is of type number
+	(excluding variables of value 'NaN')*/
+    if (typeof enteredDate === 'number' && !isNaN(enteredDate)){
```

Comment:
hi James,

I think a Boolean would fail the first conditional in my code:

(typeof enteredDate === 'number' && !isNaN(enteredDate)

Regardless I'll change the code :) - I am just not sure if you meant
that either one of these approaches
typeof enteredDate === 'number' or Number.isInteger(enteredDate) word
or whether you actually meant:

(typeof enteredDate === 'number' || Number.isInteger(enteredDate))?


k.


On Mon, Apr 25, 2016 at 9:48 AM, jamdon <notifications@github.com> wrote:

> In DateModule.js
> <https://github.com/JSCRIPT300-Spring2016/Module_2/pull/6#discussion_r60946822>
> :
>
> > +
> > +  //public API:
> > +
> > +  //setDate(): sets the internal date based on user input
> > +  //takes in: empty argument list, date object, or number)
> > +  function setDate(enteredDate) {
> > +
> > +    //check that date is set: if not, initialize date to now
> > +    if (date === null) {
> > +      _initDate();
> > +    }
> > +	
> > +	/*checks the type of the parameter enteredDate:
> > +	check if the enteredDate is of type number
> > +	(excluding variables of value 'NaN')*/
> > +    if (typeof enteredDate === 'number' && !isNaN(enteredDate)){
>
> Using !isNaN(enteredDate) will work most of the time here, but for
> certain kinds of input, will not have the expected effect. If you pass in a
> boolean value, such as true or false, for example, it will do type
> conversion. There are a couple of other approaches you could use such as typeof
> enteredDate === 'number' or Number.isInteger(enteredDate) that would be
> more reliable.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/JSCRIPT300-Spring2016/Module_2/pull/6/files/e0728d745b627deaca492234dfc8154c2aaa1b15#r60946822>
>